src/main/java/org/folio/innreach/domain/entity/InnReachTransaction.java

```diff
@@ -37,6 +48,39 @@
       "JOIN FETCH hold.pickupLocation location " +
       "WHERE t.trackingId = :trackingId AND location.id = hold.pickupLocation.id";
 
+  public static final String GET_ALL_SORTED_ASCENDING_QUERY_NAME = "InnReachTransaction.getAllSortedAscending";
+  public static final String GET_ALL_SORTED_DESCENDING_QUERY_NAME = "InnReachTransaction.getAllSortedDescending";
+  public static final String GET_ALL_SORTED_QUERY = "SELECT t FROM InnReachTransaction t JOIN FETCH t.hold hold" +
+    " JOIN FETCH t.hold holdLocal" +
+    " JOIN FETCH t.hold holdPatron" +
+    " JOIN FETCH t.hold holdItem" +
+    " JOIN FETCH holdPatron.pickupLocation locationP " +
+    " JOIN FETCH holdLocal.pickupLocation locationL " +
+    " JOIN FETCH holdItem.pickupLocation locationI " +
+    " WHERE (TYPE(holdLocal) = TransactionLocalHold AND locationL.id = holdLocal.pickupLocation.id)" +
+    " OR (TYPE(holdPatron) = TransactionPatronHold AND locationP.id = holdPatron.pickupLocation.id)" +
+    " OR (TYPE(holdItem) = TransactionItemHold AND locationI.id = holdItem.pickupLocation.id)" +
+    //list general values
+    " AND ( (:types) is null or t.type IN (:types))" +
+    " AND ( (:states) is null or t.state IN (:states))" +
+    " AND ( (:centralServerCodes) is null or t.centralServerCode IN (:centralServerCodes))" +
+    " AND ( (:patronAgencyCodes) is null or hold.patronAgencyCode IN (:patronAgencyCodes))" +
+    " AND ( (:itemAgencyCodes) is null or hold.itemAgencyCode IN (:itemAgencyCodes))" +
+    " AND ( (:centralItemTypes) is null or hold.centralItemType IN (:centralItemTypes))" +
+    //single general values
+    " AND ( :trackingId is null or t.trackingId LIKE CONCAT('%', cast(:trackingId AS text), '%'))" +
+    " AND ( :itemBarcode is null or hold.itemId LIKE :itemBarcode)" +
+    " AND ( :patronBarcode is null or hold.patronId LIKE :patronBarcode)" +
+    " AND ( :folioItemId is null or cast(hold.folioItemId AS text) LIKE CONCAT('%', cast(:folioItemId AS text), '%'))" +
+    " AND ( :folioPatronId is null or cast(hold.folioPatronId AS text) LIKE CONCAT('%', cast(:folioPatronId AS text), '%'))" +
+    //specific values
+    " AND ( :patronTypes is null OR holdItem.centralPatronType IN (:patronTypes)" +
+    " OR holdPatron.centralPatronType IN (:patronTypes) )" +" AND ( :title is null OR holdPatron.title LIKE CONCAT('%', cast(:title AS text), '%')" +
+    " OR holdLocal.title LIKE CONCAT('%', cast(:title AS text), '%'))" +
+    " AND ( :author is null OR holdPatron.author LIKE CONCAT('%', cast(:author AS text), '%')" +
+    " OR holdLocal.author LIKE CONCAT('%', cast(:author AS text), '%'))" +
+    " ORDER BY :orderBy";
```

Comment:
Consider to use Spring JPA `CriteriaQuery`, this will simplify things a lot. It supports sorting by the way. With moving to use criteria query I would also suggest wrapping filter/sort parameters to DTO(s).